hw/ip/keymgr/doc/_index.md

```diff
@@ -0,0 +1,211 @@
+
+### Disabled
+
+`Disabled` is a terminal state where the key manager is no longer operational.
+When entering Disabled random values are used to compute a new random value.
+
+## Life Cycle Connection
+The function of the key manager is directly tied to the life cycle controller.
+During specific life cycle states, the key manager is explicitly disabled.
+
+When disabled, key manager will not any advance or generate commands and if currently in an operational state, will automatically transition to `Disabled`.
+
+
+## Commands in Each State
+During each state, there are 3 valid commands software can issue:
+*  Advance state
+*  Output generation
+*  Identity generation
+
+The software is able to select a command and trigger the key manager FSM to process one of the commands.
+If a command is valid during the current working state, it is processed and acknowledged when complete.
+If a command is invalid, the key manager FSM processes with random, dummy data, but does not update working state or relevant output registers.
+
+For each valid command, a set of inputs are selected and sequenced to the KMAC module.
+
+## Generating Output Key
+The generate output command is composed of 2 options
+*  Generate output key for software
+*  Generate output key for hardware
+
+The hardware command is meant specifically for symmetric side load use cases.
+When this command is issued, the output of the KMAC invocation is not stored in software visible registers, but instead in hardware registers that directly output to symmetric primitives such as AES, HMAC and KMAC.
+
+## KMAC Operations
+All invoked KMAC operations expect the key in two-shares.
+This means the working states, even though functionally 256b, are maintained as 512b.
+
+For advance-state and generate-output commands, the KMAC emitted output are also in 2-shares.
+It is up to the software and the symmetric primitives to decide if the key can be processed as shares or XOR’d to reveal the actual value.
+
+## Block Diagram
+The following is a high level block diagram of the key manager.
+
+![Key Manager Block Diagram](keymgr_block_diagram.svg)
+
+## Design Details
+
+Key manager is primarily composed of two components:
+*  keymgr_ctrl
+*  keymgr_kmac_if
+
+### Key Manager Control
+
+The key manager control block manages the working state, sideload key  updates, as well as what commands are valid in each state.
+It also handles the life cycle keymgr_en input, which helps disable the entire key manager function in the event of an escalation.
+
+![Key Manager Control Block Diagram](keymgr_control_diagram.svg)
+
+
+### KMAC Interface Control
+
+The KMAC interface control represents the bulk of key manager logic.
+Based on input from key manager control, this module selects the inputs for each given command and sequences the data to KMAC.
+For the sake of simplicity, the diagram below shows KMAC internal, however it can also be an external module.
```

Comment:
I still don't know how to look at an SVG file in a PR via the github web interface, but this comment about internal vs external seems confusing (to me), since the KMAC is external in our current design.

Reply:
@cdgori 
at the top right of this page, click `...` and click `view file`, then you will see the diagram

Reply:
ah yes, this was when i drew the diagram with internal kmac.
I've moved it out so i've also removed that comment.